src/ex1/script.js

```diff
@@ -0,0 +1,141 @@
+let taskId = 0;
+let tasksCnt = 0;
+let isAddTaskClickable = false;
+const addTaskInput = document.querySelector(".add-task-input");
+const addTaskBtn = document.querySelector(".add-task-btn");
+
+// Add task will be clickable only for a non empty task
+addTaskInput.addEventListener("input", () => {
+  if (addTaskInput.value && !isAddTaskClickable) {
+    addTaskBtn.classList.add("valid-task-btn");
+    isAddTaskClickable = true;
+  } else if (!addTaskInput.value) {
+    addTaskBtn.classList.remove("valid-task-btn");
+    isAddTaskClickable = false;
+  }
+});
+
+// Add task
+addTaskBtn.addEventListener("click", (event) => onAddTask(event));
+document.addEventListener("keypress", (event) => {
+  if (event.key === "Enter") {
+    const inputElem = document.querySelector(".add-task-input");
+    if (inputElem === document.activeElement) {
+      onAddTask(event);
+    }
+  }
+});
+
+const createTask = () => {
+  const task = document.createElement("div");
+  task.classList.add("task-container");
+  task.id = `task-${taskId}`;
+  task.innerHTML = `
+    <div class="task-checkbox">
+        <input type="checkbox" id="input-${taskId}" />
+        <label for="input-${taskId}" />
+    </div>
+    <div class="task-txt-container">
+        <p class="task-txt" id="task-txt-${taskId}" >${addTaskInput.value}</p>
+    </div>
+    <button id="trash-button-${taskId}" class="trash">
```

Comment:
not all elements in the DOM should have an id or class, only elements that you would like to set CSS attribute or refer to them via JS

Reply:
I know, but I do use all of them (classes/ids) in other places in my code. The only 2 that I don't use are path and svg ids, which were added for debugging purposes. Removed them in Ex2.